src/workerd/api/streams/standard.c++

```diff
@@ -3267,102 +3223,74 @@ private:
 };
 }  // namespace
 
-jsg::Promise<kj::Array<byte>> ReadableStreamJsController::readAllBytes(
+template <typename T>
+jsg::Promise<T> ReadableStreamJsController::readAll(
     jsg::Lock& js,
     uint64_t limit) {
   if (isLockedToReader()) {
-    return js.rejectedPromise<kj::Array<byte>>(KJ_EXCEPTION(FAILED,
+    return js.rejectedPromise<T>(KJ_EXCEPTION(FAILED,
         "jsg.TypeError: This ReadableStream is currently locked to a reader."));
   }
   disturbed = true;
+
+  // This operation leaves the stream locked and disturbed. The loop will read until
+  // the stream is closed or errored. If the limit is reached, the loop will error.
+
+  const auto readAll = [&]() -> jsg::Promise<T> {
```

Comment:
this def. deserves a separate method: it has two nested lambdas!

Reply:
Given that this already spread out over several other methods (e.g. `readAllText(...)` and `readAllBytes(...)` each call into this... My fear is that things are already spread out a bit too much, making understanding the code flow a bit difficult. I prefer to keep this as is for readability. I will explicitly list out the captures for sure tho.